app/controllers/experiments_controller.rb

```diff
@@ -76,16 +76,24 @@ def update
       @experiment.touch(:workflowimg_updated_at)
       flash[:success] = t('experiments.update.success_flash',
                           experiment: @experiment.name)
+
       respond_to do |format|
         format.json do
           render json: {}, status: :ok
         end
+        format.html do
+          redirect_to project_path(@experiment.project)
+        end
       end
     else
+      flash[:alert] = t('experiments.archive.restore_flash')
```

Comment:
Here also other update experiment errors can occur, not just when restoring archived experiment. You can make a check if restoration is in progress, and based on that show flash error.